services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableQueryManagerImpl.java

```diff
@@ -124,6 +129,57 @@ public QueryResultBundle querySinglePage(ProgressCallback progressCallback, User
 
 	}
 
+	/**
+	 * Combined sql query is basic input query combined with all additional filter.
+	 * createCombinedSql includes the following:
+	 * <ol>
+	 * <li>Parse the query SQL string, and identify the tableId.</li>
+	 * <li>Build SqlQuery with additional filter, offset, limit and sortList.</li>
+	 * <li>Add selected facet to query</li>
+	 * <li>Parse again the combined sql to replace selectList with original select list</li>
+	 * </ol>
+	 *
+	 * @param user
+	 * @param query
+	 * @param returnCombinedSql
+	 * @return
+	 *
+	 */
+	String createCombinedSql(UserInfo user, Query query, boolean returnCombinedSql) {
+		if(!returnCombinedSql){
+			return null;
+		}
+
+		QuerySpecification model = parserQuery(query.getSql());
+		// SqlQuery changes the selectList of query so keep the original selected list.
+		SelectList selectListFromUser = model.getSelectList();
+		// We now have the table's ID.
+		String tableId = model.getSingleTableName().orElseThrow(TableConstants.JOIN_NOT_SUPPORTED_IN_THIS_CONTEXT);
+		IdAndVersion idAndVersion = IdAndVersion.parse(tableId);
+		IndexDescription indexDescription = tableManagerSupport.getIndexDescription(idAndVersion);
+		SqlQuery combinedSql = new SqlQueryBuilder(model, user.getId())
+				.schemaProvider(tableManagerSupport)
+				.overrideOffset(query.getOffset())
+				.overrideLimit(query.getLimit())
+				.indexDescription(indexDescription)
+				.selectedFacets(query.getSelectedFacets())
+				.sortList(query.getSort())
+				.additionalFilters(query.getAdditionalFilters()).build();
+
+		//SqlQuery does not include facetModel, So add selected facet to query.
+		FacetModel facetModel = new FacetModel(query.getSelectedFacets(), combinedSql, query.getSelectedFacets() != null);

```

Comment:
We want to refactor the facet model to be included in the SqlQuery.  We do not know how much work that will be.